doc/source/enhancingperf.rst

```diff
@@ -11,14 +11,147 @@
    import pandas as pd
    pd.options.display.max_rows=15
 
+   import sys
    import os
    import csv
 
+   plt.style.use('default')
+
 
 *********************
 Enhancing Performance
 *********************
 
+.. _enhancingperf.overview:
+
+
+Pandas, for most use cases, will be fast; however, there are a few
+anti-patterns to avoid. The following is a guide on achieving better
+performance with pandas. These are the low-hanging optimizations
+that should be considered before rewriting your code in Cython or to use Numba
+
+.. ipython:: python
+
+Overview of Common Optimizations
+----------------------------------------
+
+1) Using pandas when it is the wrong tool.
+
+   * At least not for things it's not meant for.
+   * Pandas is very fast at joins, reindex, factorization
+   * Not as great at, say, matrix multiplications or problems that aren't vectorizable
+
+
+
+2) Another optimization is to make sure to use native types instead of python types.
+
+.. ipython:: python
+
+   s1 = pd.Series(range(10000), dtype=object)
+   s2 = pd.Series(range(10000), dtype=np.int64)
+
+.. ipython:: python
+
+   %timeit s1.sum()
+
+.. ipython:: python
+
+   %timeit s2.sum()
+
+
+  NumPy int64 datatype arrays are much faster than the python object version.
+
+  For other datatypes:
+
+     * Strings - This is usually unavoidable. Pandas 2 will have a specialized
```

Comment:
We shouldn't talk yet about pandas 2 here IMO (or at least not start with it). This should mainly explain how to currently deal with strings.

I would explain how strings are handles in pandas -> uses object dtype. And then mention Categoricals as a way to speed up certain use cases when having a string column with many re-occuring strings.